llvm/lib/Target/X86/Disassembler/X86Disassembler.cpp

```diff
@@ -474,6 +483,23 @@ static int readPrefixes(struct InternalInstruction *insn) {
                                   insn->vectorExtensionPrefix[1],
                                   insn->vectorExtensionPrefix[2]));
     }
+  } else if (isREX2(insn, byte)) {
+    uint8_t byte1;
+    if (peek(insn, byte1)) {
+      LLVM_DEBUG(dbgs() << "Couldn't read second byte of REX2");
+      return -1;
+    }
+    insn->rex2ExtensionPrefix[0] = byte;
```

Comment:
Why do we need this since it's a constant?

Reply:
b/c we need to check there is rex2 prefix later. If not `insn->rex2ExtensionPrefix[0]`,  then we need `insn->isrex2`. They're similar.

Reply:
Can we directly compare payload with 0? I don't think there's rex2 prefix with all payload 0.

Reply:
We have rex2 prefix with payload 0 for two cases 
1. jmpabs
2. pseudo {rex2}